src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -3664,5 +3815,1138 @@ interface I
+                class Program
+                {
+                    static void Main()
+                    {
+                        var c = new C { P3 = 3, P4 = 4 };
+                        Console.WriteLine((c.P3, c.P4));
+                        foreach (var field in typeof(C).GetFields(BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static))
+                            Console.WriteLine("{0}", field.Name);
+                    }
+                }
+                """;
+            var comp = CreateCompilation(
+                [sourceA, sourceB],
+                parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion),
+                options: TestOptions.ReleaseExe,
+                targetFramework: GetTargetFramework(useInit));
+
+            if (languageVersion == LanguageVersion.CSharp13)
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (4,27): error CS8652: The feature 'field keyword' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     public partial object P3 { get; set { } }
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "P3").WithArguments("field keyword").WithLocation(4, 27),
+                    // (6,27): error CS8652: The feature 'field keyword' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     public partial object P4 { get => null; set; }
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "P4").WithArguments("field keyword").WithLocation(6, 27));
+            }
+            else
+            {
+                CompileAndVerify(
+                    comp,
+                    verify: Verification.Skipped,
+                    expectedOutput: IncludeExpectedOutput(useInit, """
+                    (, )
+                    <P3>k__BackingField
+                    <P4>k__BackingField
+                    """));
+            }
+
+            var containingType = comp.GetMember<NamedTypeSymbol>("C");
+            var actualFields = containingType.GetMembers().OfType<FieldSymbol>().ToTestDisplayStrings();
+            var expectedFields = new[]
+            {
+                "System.Object C.<P3>k__BackingField",
+                "System.Object C.<P4>k__BackingField",
+            };
+            AssertEx.Equal(expectedFields, actualFields);
+
+            var actualProperties = containingType.GetMembers().OfType<PropertySymbol>().ToArray();
+            Assert.Equal(2, actualProperties.Length);
+            Assert.True(actualProperties[0] is SourcePropertySymbol { Name: "P3", IsPartialDefinition: true, IsAutoProperty: true, UsesFieldKeyword: false, BackingField: { } });
+            Assert.True(actualProperties[1] is SourcePropertySymbol { Name: "P4", IsPartialDefinition: true, IsAutoProperty: true, UsesFieldKeyword: false, BackingField: { } });
```

Comment:
It would be good to also see verification of the backing field across the parts. e.g. whether the exact same symbol instance is expected to be used for the BackingField of both partial property parts.